src/app/deposit/deposit-dialog/deposit-dialog.component.ts

```diff
@@ -107,13 +107,23 @@ export class DepositDialogComponent implements OnInit {
    * to have a safety margin that will surpass the minimum allowed investment value.
    * @private
    */
-  private adjustDepositAmount(amount: StablecoinBigNumber, min?: StablecoinBigNumber): StablecoinBigNumber {
-    if (!min) return amount
+  private adjustDepositAmount(
+    amount: StablecoinBigNumber, min?: StablecoinBigNumber,
+  ): StablecoinBigNumber {
+    if (!min || min.lt(constants.Zero)) return amount
 
-    const safetyMargin = this.conversion.toStablecoin('1')
-    const shouldAdjust = amount.lt(min.add(safetyMargin))
+    const scaleValue = 1.0005 // proposed by Ramp team
+    const minSafe = this.conversion.scale(min, scaleValue)
+    const shouldAdjust = amount.lt(minSafe)
 
-    return shouldAdjust ? amount.add(safetyMargin) : amount
+    return shouldAdjust ? this.addSafetyMargin(min, scaleValue) : amount
```

Comment:
You could just use ceil value of `minSafe`, no need to scale the same value two times.
```suggestion
    return shouldAdjust ? this.ceilValue(minSafe, scaleValue) : amount
  }

  private ceilValue(value: StablecoinBigNumber) {
    const trimmed = this.conversion.trim(value, 'stablecoin')
    return trimmed.add(this.conversion.toStablecoin('1'))
  }
```